akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -137,6 +143,91 @@ object GraphStageLogic {
   private object DoNothing extends (() ⇒ Unit) {
     def apply(): Unit = ()
   }
+
+  trait MinimalDeathWatch {
+    this: MinimalActorRef ⇒
+    private var watchedBy = ActorCell.emptyActorRefSet
```

Comment:
Ouch, of course. Must have thought somehow that it loops through async input, but of course doesn't will fix.